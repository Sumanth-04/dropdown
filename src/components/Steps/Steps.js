import React, { Component } from "react";
import { default as RcSteps } from "rc-steps";
import Button from "../Button/index";
import "./Step.scss";

class Steps extends Component {
  constructor() {
    super();
    this.state = {
      current: 0,
      last: 0,
    };
    this.next = this.next.bind(this);
    this.submit = this.submit.bind(this);
    this.previous = this.previous.bind(this);
  }
  next() {
    this.setState({ current: this.state.current + 1 });
  }
  previous() {
    this.state.current === 0
      ? this.setState({ current: this.state.current })
      : this.setState({ current: this.state.current - 1 });
  }
  submit(e) {
    e.preventDefault();
    console.log("submit");
  }
  render() {
    var flag = this.props.onStepClick;
    var arrOfchildrenProps = [];
    for (let i = 0; i < this.props.children.length; i++) {
      arrOfchildrenProps[i] = this.props.children[i].props;
    }
    var last = arrOfchildrenProps.length;
    var divs = [];
    for (let i = 0; i < arrOfchildrenProps.length; i++) {
      divs.push(
        <RcSteps.Step
          key={i}
          title={arrOfchildrenProps[i].title}
          subTitle={arrOfchildrenProps[i].subTitle}
          onStepClick={
            flag
              ? () => {
                  this.setState({ current: i });
                }
              : undefined
          }
          description={
            this.state.current === i
              ? arrOfchildrenProps[i].description
                ? arrOfchildrenProps[i].description
                : arrOfchildrenProps[i].children
              : undefined
          }
        ></RcSteps.Step>
      );
    }

    if (last - 1 === this.state.current) {
      return (
        <div>
          <RcSteps current={this.state.current}>{divs}</RcSteps>
          <div style={{ paddingLeft: "20%",paddingTop:'25px' }}>
            <Button type="primary" size="small" onClick={this.props.submit}>
              Submit
            </Button>{" "}
            <Button type="secondary" size="small" onClick={this.previous}>
              Back
            </Button>
          </div>
        </div>
      );
    } else {
      return (
        <div>
          <RcSteps current={this.state.current}>{divs}</RcSteps>
          <div style={{ paddingLeft: "20%" ,paddingTop:'25px' }}>
            <Button type="secondary" size="small" onClick={this.next}>
              Next
            </Button>{" "}
            <Button type="secondary" size="small" onClick={this.previous}>
              Back
            </Button>
          </div>
        </div>
      );
    }
  }
}

Steps.defaultProps = {
  onStepClick: true,
};

export default Steps;
