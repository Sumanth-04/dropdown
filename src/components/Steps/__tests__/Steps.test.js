import React from "react";
import renderer from "react-test-renderer";
import Steps from "../Steps";

it("Steps should have a single child or multiple children and description or children in each child is compulsary(Use div tag)", () => {
  const tree = renderer
    .create(
      <Steps className="Steps">
        <div
          title="Personal Details"
          subTitle="Subtitle"
          description={
            <div>
              <input type="text" placeholder="First Name" />Enter First Name
              <input type="text" placeholder="Last Name" />Enter Last Name 
            </div>
          }
        ></div>
        <div
          title="Professional Details"
          subTitle="Subtitle"
          description={
            <div>
              <input type="text" placeholder=" Name" />
              <input type="text" placeholder=" Name" />
            </div>
          }
        ></div>
        <div title="Social" subTitle="Subtitle">
          <input type="text" placeholder="First Name" />
          <input type="text" placeholder="Second Name" />
        </div>
      </Steps>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
