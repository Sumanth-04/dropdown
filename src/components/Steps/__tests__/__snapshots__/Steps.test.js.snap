// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should have a single child or multiple children and description or children in each child is compulsary(Use div tag) 1`] = `
<div>
  <div
    className="rc-steps rc-steps-horizontal rc-steps-label-horizontal"
    style={Object {}}
  >
    <div
      className="rc-steps-item rc-steps-item-process rc-steps-item-active"
      style={Object {}}
    >
      <div
        className="rc-steps-item-container"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="rc-steps-item-tail"
        />
        <div
          className="rc-steps-item-icon"
        >
          <span
            className="rc-steps-icon"
          >
            1
          </span>
        </div>
        <div
          className="rc-steps-item-content"
        >
          <div
            className="rc-steps-item-title"
          >
            Personal Details
            <div
              className="rc-steps-item-subtitle"
              title="Subtitle"
            >
              Subtitle
            </div>
          </div>
          <div
            className="rc-steps-item-description"
          >
            <div>
              <input
                placeholder="First Name"
                type="text"
              />
              Enter First Name
              <input
                placeholder="Last Name"
                type="text"
              />
              Enter Last Name
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rc-steps-item rc-steps-item-wait"
      style={Object {}}
    >
      <div
        className="rc-steps-item-container"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="rc-steps-item-tail"
        />
        <div
          className="rc-steps-item-icon"
        >
          <span
            className="rc-steps-icon"
          >
            2
          </span>
        </div>
        <div
          className="rc-steps-item-content"
        >
          <div
            className="rc-steps-item-title"
          >
            Professional Details
            <div
              className="rc-steps-item-subtitle"
              title="Subtitle"
            >
              Subtitle
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="rc-steps-item rc-steps-item-wait"
      style={Object {}}
    >
      <div
        className="rc-steps-item-container"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="rc-steps-item-tail"
        />
        <div
          className="rc-steps-item-icon"
        >
          <span
            className="rc-steps-icon"
          >
            3
          </span>
        </div>
        <div
          className="rc-steps-item-content"
        >
          <div
            className="rc-steps-item-title"
          >
            Social
            <div
              className="rc-steps-item-subtitle"
              title="Subtitle"
            >
              Subtitle
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingLeft": "20%",
        "paddingTop": "25px",
      }
    }
  >
    <button
      className="rd-button rd-secondary small rd- undefined "
      label=""
      onClick={[Function]}
      size="small"
      style={Object {}}
      type="secondary"
    >
      Next
       
      
       
    </button>
     
    <button
      className="rd-button rd-secondary small rd- undefined "
      label=""
      onClick={[Function]}
      size="small"
      style={Object {}}
      type="secondary"
    >
      Back
       
      
       
    </button>
  </div>
</div>
`;
